src/v/kafka/server/handlers/fetch.cc

```diff
@@ -702,6 +711,186 @@ class parallel_fetch_plan_executor final : public fetch_plan_executor::impl {
     }
 };
 
+class pid_controller {
+    // 0.1234 * scaling_factor = 1234
+    static constexpr int64_t scaling_factor = 10000;
+    static constexpr auto duration_unit = 1us;
+    static constexpr auto sample_rate = 250us;
+
+public:
+    pid_controller()
+      : _last_sampled_time{ss::sched_clock::now()}
+      , _p_coeff{config::shard_local_cfg().fetch_pid_p_coeff()}
+      , _i_coeff{config::shard_local_cfg().fetch_pid_i_coeff()}
+      , _d_coeff{config::shard_local_cfg().fetch_pid_d_coeff()}
+      , _target_fetch_sg_util{(
+          int64_t)(scaling_factor
+                   * config::shard_local_cfg().fetch_pid_target_fetch_sg())}
+      , _max_delay{config::shard_local_cfg().fetch_pid_max_debounce()}
+      , _p_coeff_binding{config::shard_local_cfg().fetch_pid_p_coeff.bind()}
+      , _i_coeff_binding{config::shard_local_cfg().fetch_pid_i_coeff.bind()}
+      , _d_coeff_binding{config::shard_local_cfg().fetch_pid_d_coeff.bind()}
+      , _target_fetch_sg_util_binding{config::shard_local_cfg()
+                                        .fetch_pid_target_fetch_sg.bind()}
+      , _max_delay_binding{
```

Comment:
FWIW when you have a long list of member initializers like this, and the only constructor takes no arguments, you might consider just initializing stuff inline where the member is declared, saves a lot of duplication and makes it easier to check that everything is initialized.